zerver/webhooks/gitea/view.py

```diff
@@ -15,21 +17,34 @@
 fixture_to_headers = get_http_headers_from_filename("HTTP_X_GITEA_EVENT")
 
 
-def format_pull_request_event(payload: Dict[str, Any], include_title: bool = False) -> str:
+def format_pull_request_event(payload: WildValue, include_title: bool = False) -> str:
     assignee = payload["pull_request"]["assignee"]
-    data = {
-        "user_name": payload["pull_request"]["user"]["username"],
-        "action": payload["action"],
-        "url": payload["pull_request"]["html_url"],
-        "number": payload["pull_request"]["number"],
-        "target_branch": payload["pull_request"]["head"]["ref"],
-        "base_branch": payload["pull_request"]["base"]["ref"],
-        "title": payload["pull_request"]["title"] if include_title else None,
-        "assignee": assignee["login"] if assignee else None,
+    DataType = TypedDict(
+        "DataType",
+        {
+            "user_name": str,
+            "action": str,
+            "url": str,
+            "number": int,
+            "target_branch": str,
+            "base_branch": str,
+            "title": Optional[str],
+            "assignee": Optional[str],
+        },
+    )
+    data: DataType = {
+        "user_name": payload["pull_request"]["user"]["username"].tame(check_string),
+        "action": payload["action"].tame(check_string),
+        "url": payload["pull_request"]["html_url"].tame(check_string),
+        "number": payload["pull_request"]["number"].tame(check_int),
+        "target_branch": payload["pull_request"]["head"]["ref"].tame(check_string),
+        "base_branch": payload["pull_request"]["base"]["ref"].tame(check_string),
+        "title": payload["pull_request"]["title"].tame(check_string) if include_title else None,
+        "assignee": assignee["login"].tame(check_string) if assignee else None,
     }
 
-    if payload["pull_request"]["merged"]:
-        data["user_name"] = payload["pull_request"]["merged_by"]["username"]
+    if payload["pull_request"]["merged"].tame(check_bool):
+        data["user_name"] = payload["pull_request"]["merged_by"]["username"].tame(check_string)
         data["action"] = "merged"
 
     return get_pull_request_event_message(**data)
```

Comment:
There’s no need to build a `TypedDict` just to call a function. Either explode it to individual variables, or pass the keyword arguments directly:

```python
    pull_request = payload["pull_request"]
    merged = pull_request["merged"].tame(check_bool)

    return get_pull_request_event_message(
        user_name=pull_request["merged_by" if merged else "user"]["username"].tame(check_string),
        action="merged" if merged else payload["action"].tame(check_string),
        url=pull_request["html_url"].tame(check_string),
        number=pull_request["number"].tame(check_int),
        target_branch=pull_request["head"]["ref"].tame(check_string),
        base_branch=pull_request["base"]["ref"].tame(check_string),
        title=pull_request["title"].tame(check_string) if include_title else None,
        assignee=assignee["login"].tame(check_string) if assignee else None,
    )
```

Reply:
Fair enough, will do!